### PR TITLE
feat(formula): supplement FORMULATEXT formula

### PR DESCRIPTION
**Pull request by wpxp123456**
- [ ] FORMULATEXT

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/engine-formula/src/functions/base-function.ts [hunk 1]
```diff
@@ -148,6 +155,14 @@ export class BaseFunction {
         this._sheetNameMap = sheetNameMap;
     }
 
+    getFormulaDataModel() {
```
Comment (wzhudev) [inline packages/engine-formula/src/functions/base-function.ts:158]: 不要把从 constructor 获取到的外部变量再通过 getter 暴露出去，多使用 protected 类型的属性。

File: packages/engine-formula/src/functions/base-function.ts [hunk 2]
```diff
@@ -43,6 +44,7 @@ export class BaseFunction {
     private _locale: LocaleType;
     private _sheetOrder: string[];
     private _sheetNameMap: { [sheetId: string]: string };
+    private _formulaDataModel: FormulaDataModel;
```
Comment (wzhudev) [inline packages/engine-formula/src/functions/base-function.ts:47]:
```suggestion
    protected _formulaDataModel: Nullable<FormulaDataModel>;
```

**Review by wzhudev — COMMENTED** [review]
Overall LGTM. 需要修改下代码以提升封装性

File: packages/engine-formula/src/functions/base-function.ts (hunk 1 above)
Comment (wpxp123456) [reply packages/engine-formula/src/functions/base-function.ts:158]: ok

**Review by Dushusir — APPROVED** [review]

**Pull request merged**